### PR TITLE
feat: Endpoint for checking if an email is available

### LINKED ISSUE #4939: Allow checking whether an email is registered
**Is your feature request related to a problem? Please describe.**

Users API should allow checking whether an email is registered.

**Describe the solution you'd like**

A POST request requiring only an email address should respond whether the email is registered or not.

**Additional context**

Orga App issue: [#971](https://github.com/fossasia/open-event-orga-app/issues/971)

### PR DESCRIPTION
**Pull request by mayank8318**
Fixes #4939 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Add an endpoint to check if the email is available or not

#### Changes proposed in this pull request:
- Add a miscellaneous endpoint to check if the email is available or not
- Return true if it is available, false otherwise

Email already present-
screenshot from 2018-07-07 11-52-36

Email available-
screenshot from 2018-07-07 11-53-05

Error in the request-
screenshot from 2018-07-09 11-29-19

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -131,6 +132,7 @@ def create_app():
         app.register_blueprint(celery_routes)
         app.register_blueprint(auth_routes)
         app.register_blueprint(event_statistics)
+        app.register_blueprint(user_misc_routes)
```
Comment (srv-twry) [inline app/__init__.py]: Why this ?

File: app/api/users.py [hunk 2]
```diff
@@ -20,15 +21,18 @@
...
+from app.api.helpers.exceptions import ConflictException, UnprocessableEntity
+from app.api.helpers.db import safe_query, get_count
 from app.models.user import User
 from app.models.users_events_role import UsersEventsRoles
 from app.models.ticket_holder import TicketHolder
 from app.api.helpers.exceptions import ForbiddenException
 from app.api.helpers.permission_manager import has_access
 
 
+user_misc_routes = Blueprint('user_misc', __name__, url_prefix='/v1')
```
Comment (srv-twry) [inline app/api/users.py]: We used to do this in v1. Please check ```app/api/__init__.py``` file. We declare endpoints in that file now.

File: app/api/users.py [hunk 3]
```diff
@@ -215,3 +219,19 @@ class UserRelationship(ResourceRelationship):
     schema = UserSchema
     data_layer = {'session': db.session,
                   'model': User}
+
+
+@user_misc_routes.route('/users/checkEmail', methods=['POST'])
```
Comment (srv-twry) [inline app/api/users.py]: this won't be need then.

File: app/api/users.py [hunk 4]
```diff
@@ -215,3 +219,19 @@ class UserRelationship(ResourceRelationship):
...
+    email = request.json.get('email', None)
+    if email:
+        if get_count(db.session.query(User).filter_by(email=email)):
+            return jsonify(
+                result="False"
+            )
+        else:
+            return jsonify(
+                result="True"
+            )
```
Comment (srv-twry) [inline app/api/users.py]: Declare a schema and use that. See how the `ChargesLayer` is used without any model(In app/api/orders.py). Similarly you can create a `UserLayer`.

Comment (srv-twry) [reply app/api/users.py]: It seems like this will be an overkill. You can leave it as it is. @mayank8318

Comment (mayank8318) [reply app/api/users.py]: That seems like an overkill for this IMO

**Comment by srv-twry** [comment]
@mayank8318 The error response is not correct as shown in the screenshot.

**Comment by mayank8318** [comment]
@srv-twry Updated :)

**Pull request merged**